src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php

```diff
@@ -94,4 +95,22 @@ private function loadResources()
             unset($this->resourceFiles[$key]);
         }
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function warmUp($cacheDir)
+    {
+        if (null !== $this->options['cache_dir']) {
+            if (null !== $this->locale) {
+                $this->loadCatalogue($this->locale);
```

Comment:
@aitboudad I guess it was here for a good reason. Seems it avoid a kind of infinite (or very long) loop somewhere.

Results for ```time php app/console cache:clear``` in prod or dev environments :

Env |Type|Before|After
----|----|------|---------
prod|real|0m6.315s|1m48.515s
    |user|0m3.224s|1m28.898s
    |sys |0m0.696s|0m9.677s
----|----|--------|---------
dev |real|0m3.364s|2m0.263s
    |user|0m2.228s|1m34.026s
    |sys |0m0.448s|0m13.129s

I don't have time to investigate more. I'll add some tests and then I'll be done for this PR.